mint_lib/src/mod_info.rs

```diff
@@ -15,30 +19,30 @@ pub struct ModioTags {
     pub approval_status: ApprovalStatus,
 }
 
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub enum RequiredStatus {
     RequiredByAll,
     Optional,
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Hash)]
 pub enum ApprovalStatus {
     Verified,
     Approved,
     Sandbox,
 }
 
 /// Whether a mod can be resolved by clients or not
-#[derive(Debug, Clone, Eq, Ord, PartialEq, PartialOrd, Hash)]
+#[derive(Debug, Clone, Eq, Ord, PartialEq, PartialOrd, Hash, Serialize, Deserialize)]
 pub enum ResolvableStatus {
     Unresolvable(String),
     Resolvable,
 }
 
 /// Returned from ModStore
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct ModInfo {
-    pub provider: &'static str,
+    pub provider: String,
```

Comment:
Is this type change only here because `ModInfo` is being used as a key for `GameplayAffecting` status?

Reply:
Yes. I will revert it back.